keyboards/yampad/keymaps/default/keymap.c

```diff
@@ -0,0 +1,140 @@
+ * |DEL |END |PGDN|    |
+ * |----|----|----|    |
+ * |    |    |    |    |
+ * |----|----|----|----|
+ * |    | UP |    |    |
+ * |----|----|----|    |
+ * |LEFT|DOWN|RIGH|    |
+ * `-------------------'
+ */
+  [_NV] = LAYOUT(
+    KC_INS,   KC_HOME,  KC_PGUP,   TG(_NV), \
+    KC_DEL,   KC_END,   KC_PGDN,            \
+    XXXXXXX,  XXXXXXX,  XXXXXXX,   XXXXXXX, \
+    XXXXXXX,  KC_UP,    XXXXXXX,            \
+    KC_LEFT,  KC_DOWN,  KC_RGHT,   XXXXXXX
+  ),
+
+/* Keymap _FN: RGB Function Layer
+ * ,-------------------.
+ * |RMOD|RGBP|RTOG| FN |
+ * |----|----|----|----|
+ * |HUD |HUI |    |    |
+ * |----|----|----|    |
+ * |SAD |SAI |    |    |
+ * |----|----|----|----|
+ * |VAD |VAS |    |    |
+ * |----|----|----|    |
+ * |RST |    |    |    |
+ * `-------------------'
+ */
+  [_FN] = LAYOUT(
+    RGB_MOD,  RGB_M_P,  RGB_TOG,   _______, \
+    RGB_HUD,  RGB_HUI,  XXXXXXX,            \
+    RGB_SAD,  RGB_SAI,  XXXXXXX,   XXXXXXX, \
+    RGB_VAD,  RGB_VAI,  XXXXXXX,            \
+    RESET,    XXXXXXX,  XXXXXXX,   XXXXXXX
+  ),
+};
+
+bool process_record_user(uint16_t keycode, keyrecord_t *record) {
+  switch (keycode) {
+    case KC_DBL0:
+      if (record->event.pressed) {
+        SEND_STRING("00");
+      } else {
+        // when keycode KC_DBL0 is released
+      }
+      break;
+
+  }
+  return true;
+};
+
+#ifdef OLED_DRIVER_ENABLE
+
+oled_rotation_t oled_init_user(oled_rotation_t rotation) {
+    return OLED_ROTATION_270;  // flips the display 270 degrees
+}
+
+void oled_task_user(void) {
+  // Host Keyboard Layer Status
+  oled_write_P(PSTR("Layer"), false);
+  switch (biton32(layer_state)) {
+    case _BL:
+      oled_write_ln_P(PSTR(" BAS"), false);
+      break;
+    case _NV:
+      oled_write_ln_P(PSTR(" NAV"), false);
+      break;
+    case _FN:
+      oled_write_ln_P(PSTR(" RGB"), false);
+      break;
+    default:
+      // Or use the write_ln shortcut over adding '\n' to the end of your string
+      oled_write_ln_P(PSTR(" UND"), false);
+  }
+
+  // Host Keyboard LED Status
+  uint8_t led_usb_state = host_keyboard_leds();
+  oled_write_P(PSTR("-----"), false);
+  oled_write_P(PSTR("Stats"), false);
+  oled_write_P(led_usb_state & (1<<USB_LED_NUM_LOCK) ? PSTR("num:*") : PSTR("num:."), false);
+  oled_write_P(led_usb_state & (1<<USB_LED_CAPS_LOCK) ? PSTR("cap:*") : PSTR("cap:."), false);
+  oled_write_P(led_usb_state & (1<<USB_LED_SCROLL_LOCK) ? PSTR("scr:*") : PSTR("scr:."), false);
+
+  // Host Keyboard RGB backlight status
+  oled_write_P(PSTR("-----"), false);
+  oled_write_P(PSTR("Light"), false);
+
+  static char led_buf[30];
+  snprintf(led_buf, sizeof(led_buf) - 1, "RGB:%cM: %2d\nh: %2ds: %2dv: %2d\n",
+      rgblight_config.enable ? '*' : '.', (uint8_t)rgblight_config.mode,
+      (uint8_t)(rgblight_get_hue() / RGBLIGHT_HUE_STEP),
+      (uint8_t)(rgblight_get_sat() / RGBLIGHT_SAT_STEP),
+      (uint8_t)(rgblight_get_val() / RGBLIGHT_VAL_STEP));
+  oled_write(led_buf, false);
+}
+#endif
```

Comment:
It may be with moving this to the keyboard's C file but set weakly (eg `__attribute__((weak))` before the function names, so it can be overridden by the user)

Reply:
Hi, not sure exactly what you mean. C is not my language of choice...